server/sonar-server/src/main/java/org/sonar/server/component/index/ComponentIndex.java

```diff
@@ -49,42 +53,59 @@
 import static org.sonar.server.component.index.ComponentIndexDefinition.FIELD_NAME;
 import static org.sonar.server.component.index.ComponentIndexDefinition.FIELD_QUALIFIER;
 import static org.sonar.server.component.index.ComponentIndexDefinition.INDEX_TYPE_COMPONENT;
+import static org.sonar.server.component.index.ComponentIndexDefinition.NAME_ANALYZERS;
 
 public class ComponentIndex {
 
   private static final String FILTERS_AGGREGATION_NAME = "filters";
   private static final String DOCS_AGGREGATION_NAME = "docs";
 
   private final EsClient client;
+  private final FavoriteFinder favoriteFinder;
```

Comment:
IMO it's not the responsibility of ComponentIndex to load user favorites. That leads to need to propagate data in ComponentIndexResults#favorites whereas it does not relate at all to Elasticsearch index. 
The list of favorites should be provided by the caller (the WS).